aspnetcore/blazor/state-management/server.md

```diff
@@ -0,0 +1,239 @@
+    {
+        Items ??= await LoadItemsAsync();
+    }
+}
+```
+
+To persist State for scoped services, annotate service properties with `[SupplyParameterFromPersistentComponentState]`, add the service to the service collection, and call the <xref:Microsoft.Extensions.DependencyInjection.RazorComponentsRazorComponentBuilderExtensions.RegisterPersistentService%2A> extension method with the service:
+
+```csharp
+public class CustomUserService
+{
+    [SupplyParameterFromPersistentComponentState]
+    public string UserData { get; set; }
+}
+
+services.AddScoped<CustomUserService>();
+
+services.AddRazorComponents()
+  .AddInteractiveServerComponents()
+  .RegisterPersistentService<CustomUserService>(RenderMode.InteractiveAuto);
+```
+
+> [NOTE]
+> The preceding example persists `UserData` state when the service is used in component prerendering for both Interactive Server and Interactive WebAssembly rendering because `RenderMode.InteractiveAuto` is specified to `RegisterPersistentService`. However, circuit state persistence is only available for the **Interactive Server** render mode.
+
+To handle distributed state persistence (and to act as the default state persistence mechanism when configured), assign a [`HybridCache`](xref:performance/caching/overview#hybridcache) (API: <xref:Microsoft.Extensions.Caching.Hybrid.HybridCache>) to the app, which configures its own persistence period (`PersistedCircuitDistributedRetentionPeriod`, eight hours by default). `HybridCache` is used because it provides a unified approach to distributed storage that doesn't require separate packages for each storage provider.
+
+In the following example, a <xref:Microsoft.Extensions.Caching.Hybrid.HybridCache> is implemented with the [Redis](https://redis.io/) storage provider:
+
+```csharp
+services.AddHybridCache()
+    .AddRedis("{CONNECTION STRING}");
+
+services.AddRazorComponents()
+    .AddInteractiveServerComponents();
+```
+
+In the preceding example, the `{CONNECTION STRING}` placeholder represents the Redis cache connection string, which should be provided using a secure approach, such as the [Secret Manager](xref:security/app-secrets#secret-manager) tool in the Development environment or [Azure Key Vault](/azure/key-vault/) with [Azure Managed Identities](/entra/identity/managed-identities-azure-resources/overview) for Azure-deployed apps in any environment.
+
+To proactively pause and resume circuits in custom resource management scenarios, call `Blazor.pauseCircuit` and `Blazor.resumeCircuit` from a JavaScript event handler. In the following example, changes in the the visibility of the app either pause or resume the user's circuit:
+
+```javascript
+window.addEventListener('visibilitychange', () => {
+  if (document.visibilityState === 'hidden') {
+    Blazor.pauseCircuit();
+  } else if (document.visibilityState === 'visible') {
+    Blazor.resumeCircuit();
+  }
+});
+```
+
+## Declarative model for persisting state using a DI service
```

Comment:
I think state persistence for services was covered earlier in this doc (line 133). Maybe this section can be removed?